### ISSUE TITLE
Trinity container: running an MPI job with IB fails

### ISSUE THREAD

**Issue by hansthen** [clustervision/trinity#174]
**steps to reproduce:**
```
[root@login ~]# su - jane
-bash-4.2$ sbatch -N 2 --exclusive
#!/bin/bash
module load imb
module load openmpi/gcc/64/1.8.4                                    
mpirun --mca btl self,tcp,openib IMB-MPI1 pingpong
Submitted batch job 59
```
**actual output:**
```
-bash-4.2$ ssh c001 cat slurm-59.out | less
------------------------------------------------------------
A process or daemon was unable to complete a TCP connection
to another process:
  Local host:    c001
  Remote host:   c001
This is usually caused by a firewall on the remote host. Please
check that any firewall (e.g., iptables) has been disabled and
try again.
```
**expected output:**
```
-bash-4.2$ sbatch -N 2
#!/bin/bash
module load imb
module load openmpi/gcc/64/1.8.4                                    
mpirun --mca btl self,tcp IMB-MPI1 pingpong
^D
Submitted batch job 54
 benchmarks to run pingpong 
#------------------------------------------------------------
#    Intel (R) MPI Benchmarks 4.0 Update 1, MPI-1 part    
#------------------------------------------------------------
```

**Comment by hansthen** [comment]
The expected output shows that using only TCP as a point to point layer makes the system work.

Some more errors (when logged into the node).

```
-bash-4.2$ ssh c001
Warning: Permanently added 'c001,172.17.0.1' (RSA) to the list of known hosts.
Last login: Thu Apr  9 18:53:25 2015 from login-b.vc-b
-bash-4.2$ ping c001
ping: icmp open socket: Operation not permitted
-bash-4.2$ ssh c001
The authenticity of host 'c001 (172.17.0.1)' can't be established.
RSA key fingerprint is 41:c1:ac:9b:f8:02:57:61:c2:82:a1:9f:a9:1a:8d:fc.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added 'c001' (RSA) to the list of known hosts.
Last login: Thu Apr  9 19:14:44 2015 from login-b.vc-b
-bash-4.2$ 
```

See these messages for the capability to ping. This may be a completely unrelated issue, as we are capable of ssh-ing into the same node.

https://bugs.centos.org/view.php?id=8148
https://bugzilla.redhat.com/show_bug.cgi?id=1142311

```
-bash-4.2$ ibhosts
src/query_smp.c:228; can't open UMAD port ((null):0)
/usr/sbin/ibnetdiscover: iberror: failed: discover failed
```

**Comment by hansthen** [comment]
This appears to be a rights issue. As root the ibhosts command works:
```
[root@c001 /]# ibhosts
Ca	: 0xf452140300224be0 ports 1 "node002 mlx4_0"
Ca	: 0x0002c903000bbdb4 ports 1 "MT25408 ConnectX Mellanox Technologies"
Ca	: 0x0002c90300eaa290 ports 1 "node001 mlx4_0"
```

**Comment by hansthen** [comment]
This probably has to do with memory limits (as in #158). As root the command works. I updated the memory limits inside the container, but slurm does not see that.

```
john@login ~> sbatch -N 2 --exclusive 
#!/bin/bash
ulimit -l
Submitted batch job 13
john@login ~> cat slurm-13.out 
64
```

and compare

```
john@c001 ~> ulimit -l
```

**Comment by hansthen** [comment]
After (a lot of) digging, it turns out we need to set the memory limits inside the docker startup script.

**Comment by hansthen** [comment]
This alone is not enough. I got it to work with the following additions.
1. On the login node add /etc/security/limits.d/trinity.conf
```
*               -       stack           unlimited
*               -       memlock         unlimited
```
2. On the login node add /etc/sysconfig/slurm
```
ulimit -l unlimited
ulimit -s unlimited
```

**Comment by hansthen** [comment]
@abhimukh What is the best way to add files to be copied into the login image? Is there something like a rootimg location in the repo for login nodes?

**Comment by abhimukh** [comment]
@hansthen You can only pass a single file (user_data) for login node configuration. Currently it is under <repo_root>/trinitylogin/etc/trinity/login.conf

**Comment by hansthen** [comment]
**Fixed in:** 4bcce3c
